### PR TITLE
TEST: reformat all nix files with nixpkgs-fmt

### PR DESCRIPTION
**Pull request by domenkozar**
DO NOT MERGE

Testing https://github.com/NixOS/nixpkgs/issues/120832 with:

```
find . -name "*.nix" -exec $(nix-build -A nixpkgs-fmt)/bin/nixpkgs-fmt {} \;
```

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Thanks for starting this. Let us know if you need some help or some build power to test something.

**Comment by Synthetica9** [comment]
The commit message for the main commit here is crucial documentation that can't be changed afterwards; it is therefore critical to get this right. I'd propose something like:

> treewide: format with nixpkgs-fmt
> 
> If you are seeing this in git blame, you have not set up .git-blame-ignore-revs properly. To do this, you need to run the following command once:
> 
>     git config blame.ignoreRevsFile .git-blame-ignore-revs
> 
> This should work with all local tools.
> For more information see [INFO URL].

The info URL is so we can give more information in the future.

**Comment by Mindavi** [comment]
I like the general idea of doing this.

However, there are some things I don't like about the current formatting:

- I don't think comments should be reformatted; they sometimes contain nix expressions, which are now flattened by the formatter (or it should be more intelligent, but I think not formatting is better)
- I saw something where multiple values were listed like below, and the formatting got weird

packageA,
# comment
packageB,
# comment

**Comment by grahamc** [comment]
I agree it should leave comments alone. A good example is these docs:

```

  /* Specialized `assertMsg` for checking if val is one of the elements
     of a list. Useful for checking enums.
    of a list. Useful for checking enums.
     Example:
       let sslLibrary = "libressl"
       in assertOneOf "sslLibrary" sslLibrary [ "openssl" "bearssl" ]
       => false
       stderr> trace: sslLibrary must be one of "openssl", "bearssl", but is: "libressl"
    Example:
    let sslLibrary = "libressl"
    in assertOneOf "sslLibrary" sslLibrary [ "openssl" "bearssl" ]
    => false
    stderr> trace: sslLibrary must be one of "openssl", "bearssl", but is: "libressl"
     Type:
       assertOneOf :: String -> ComparableVal -> List ComparableVal -> Bool
    Type:
    assertOneOf :: String -> ComparableVal -> List ComparableVal -> Bool
  */
```

the flattening here is unfortunate.

**Comment by domenkozar** [comment]
> I like the general idea of doing this.
> 
> However, there are some things I don't like about the current formatting:
> 
> * I don't think comments should be reformatted; they sometimes contain nix expressions, which are now flattened by the formatter (or it should be more intelligent, but I think not formatting is better)
> * I saw something where multiple values were listed like below, and the formatting got weird
> 
> packageA,
> # comment
> packageB,
> # comment
> 
> It was formatted to this:
> 
> packageA
> , # comment
> packageB
> , # comment
> 
> The comments are really not great, the second thing is less of an issue, but I'd never manually format it like that.
> 
> (This was just with a brief look, haven't looked very long at this)

That looks like https://github.com/nix-community/nixpkgs-fmt/issues/243

**Comment by domenkozar** [comment]
I wanted to open this to test that whole nixpkgs/NixOS still evals. Unfortunately master moves quite fast and conflicts arise as soon as something is merged. Is there a way to run the whole eval suite locally?

File: lib/asserts.nix [hunk 1]
```diff
@@ -3,19 +3,19 @@
...
 
   /* Print a trace message if pred is false.
-     Intended to be used to augment asserts with helpful error messages.
+    Intended to be used to augment asserts with helpful error messages.
 
-     Example:
-       assertMsg false "nope"
-       => false
-       stderr> trace: nope
+    Example:
```
Comment (FRidh) [inline lib/asserts.nix:8]: Indentation of comments get lost. nixpkgs-fmt probably should ignore comments.

Comment (Synthetica9) [reply lib/asserts.nix:8]: How much work would it be to patch out the comment handling?

Comment (domenkozar) [reply lib/asserts.nix:8]: > > I like the general idea of doing this.
> > However, there are some things I don't like about the current formatting:
> > 
> > * I don't think comments should be reformatted; they sometimes contain nix expressions, which are now flattened by the formatter (or it should be more intelligent, but I think not formatting is better)
> > * I saw something where multiple values were listed like below, and the formatting got weird
> > 
> > packageA,
> > # comment
> > packageB,
> > # comment
> > It was formatted to this:
> > packageA
> > , # comment
> > packageB
> > , # comment
> > The comments are really not great, the second thing is less of an issue, but I'd never manually format it like that.
> > (This was just with a brief look, haven't looked very long at this)
> 
> That looks like [nix-community/nixpkgs-fmt#243](https://github.com/nix-community/nixpkgs-fmt/issues/243)

That's fixed in nixpkgs-fmt master. Going to see if I can use that version and test it here.

**Comment by domenkozar** [comment]
Built nixpkgs-fmt from master and ran `find . -name "*.nix" -exec /nix/store/askqcqa4mrblhpgyks60hxiph8c61haj-nixpkgs-fmt-1.2.0/bin/nixpkgs-fmt {} \;` on fresh master.

**Comment by Mindavi** [comment]
This still doesn't look quite right:

https://github.com/NixOS/nixpkgs/pull/121490/files#diff-9a742acd363e7062932805c87186c92a025bef9d215fda52b8d46631691bbe9bR54-R62

But maybe that's intended, I'm not sure. It's not something I'd do by hand though.

But overall this looks a lot better to me!

**Comment by domenkozar** [comment]
Comma at the end of at the beginning is a debate I'd like to avoid here and what's important is that we can have either as that is a lossless conversion.

Other things anyone can notice? Otherwise it seems like nixpkgs-fmt formats all files and eval checks pass. I've also tested that running it twice doesn't make any new modifications :popcorn:

**Comment by Ma27** [comment]
* Does anybody have a smart idea what tool to use for a review? :sweat_smile: `git show` in a `less` seems quite painful already ^^
* AFAICS also automagically created files are touched. Shouldn't we exclude these? (or fix the upstream tools)
* According to the labels this would cause a full stdenv rebuild (and from the changes I've seen in `pkgs/stdenv` this seems correct). Given that this is likely a candidate for staging, are we sure that this will be ready on time before the branchoff (which is on friday IIRC, cc @NixOS/release-engineers )

**Comment by zowoq** [comment]
> AFAICS also automagically created files are touched. Shouldn't we exclude these? (or fix the upstream tools)

One auto-generated file looks like it is responsible for the majority of the diff.

```
 pkgs/development/haskell-modules/hackage-packages.nix
 1 file changed, 505371 insertions(+), 249022 deletions(-)
```

**Comment by asymmetric** [comment]
> Comma at the end of at the beginning is a debate I'd like to avoid here and what's important is that we can have either way as that is a lossless conversion.

@domenkozar 

This would look better IMO:

```nix
{
  foo
  # comments about bar
, bar
}
```

I think it would be valid too? Not sure how feasible or easy though.

**Pull request closed without merging**